src/corelocation.cs

```diff
@@ -736,4 +806,30 @@ interface CLVisit : NSSecureCoding, NSCopying {
 		double HorizontalAccuracy { get; }
 	}
 #endif
+
+	[NoWatch, NoTV, NoMac, iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor] // NSInvalidArgumentException Reason: -[CLBeaconIdentityConstraint init]: unrecognized selector sent to instance 0x600001930300
+	interface CLBeaconIdentityConstraint : NSCopying, NSSecureCoding {
+
+		[Export ("initWithUUID:")]
+		IntPtr Constructor (NSUuid uuid);
+
+		[Export ("initWithUUID:major:")]
+		IntPtr Constructor (NSUuid uuid, ushort major);
+
+		[Export ("initWithUUID:major:minor:")]
+		IntPtr Constructor (NSUuid uuid, ushort major, ushort minor);
+
+		[Export ("UUID", ArgumentSemantic.Copy)]
+		NSUuid Uuid { get; }
+
+		[NullAllowed, Export ("major", ArgumentSemantic.Copy)]
+		[BindAs (typeof (short?))]
```

Comment:
When do we use `BindAs`?

Reply:
There's a few cases, the most common one is `NSNumber`, like above. Another one are to support "smart enums" (using a C# enum that maps to several `NSString` constants).

So for some API Apple prefers to use a reference type, like `NSNumber`,  over a more specific value type. Why ? mainly because they want a nullable properties

But
```csharp
NSNumber Major { get; }
```
does not tell the API consumer if it should except a `short` (like it's meant to), a `float` or even a `bool`. That does not create a very discoverable API and we can do a lot better in C# with nullable `?` types :)

Why not
```csharp
short? Major { get; }
```

Because our runtime must know this an `NSNumber` natively - e.g. it's needed to allow us to override the property in a subclass and have some ObjC code call us. So the declaration remains an `NSNumber` but we give the generator a hint to add extra (conversion) code to expose a nicer API to our developers.

Reply:
Also we have some docs about it :) https://docs.microsoft.com/en-us/xamarin/cross-platform/macios/binding/objective-c-libraries?tabs=macos#binding-nsvalue-nsnumber-and-nsstring-to-a-better-type

Reply:
Thank you both for the detailed explanation! 👍